mne/utils/numpy.py

```diff
@@ -0,0 +1,34 @@
+# Authors: The MNE-Python contributors.
+# License: BSD-3-Clause
+# Copyright the MNE-Python contributors.
+
+import io
+import os
+
+import numpy
+
+AnyFile = str | bytes | os.PathLike | io.IOBase
+
+
+def numpy_fromfile(
+    file: AnyFile, dtype: numpy.typing.DTypeLike = float, count: int = -1
+):
+    """numpy.fromfile() wrapper, handling io.BytesIO file-like streams.
+
+    Numpy requires open files to be actual files on disk, i.e., must support
+    file.fileno(), so it fails with file-like streams such as io.BytesIO().
+
+    If numpy.fromfile() fails due to no file.fileno() support, this wrapper
+    reads the required bytes from file and redirects the call to
+    numpy.frombuffer().
+
+    See https://github.com/numpy/numpy/issues/2230
```

Comment:
link to https://github.com/numpy/numpy/issues/2230#issuecomment-949795210 would be more appropriate here.